packages/belos/tpetra/example/BlockGmres/BlockGmresTpetraGaleriEx.cpp

```diff
@@ -0,0 +1,290 @@
+    bool pseudo = false;       // use pseudo-block or block gmres
+    int frequency = -1;        // frequency of status test output
+    int blocksize = 1;         // blocksize
+    int numrhs = 1;            // number of right-hand sides to solve for
+    int maxiters = -1;         // maximum number of iterations allowed per linear system
+    int maxsubspace = 50;      // maximum number of blocks the solver can use for the subspace
+    int maxrestarts = 15;      // number of restarts allowed
+    int nx = 10;               // number of discretization points in each direction
+    MT tol = 1.0e-5;           // relative residual tolerance
+    std::string ortho = "DGKS";// orthogonalization method
+
+    Teuchos::CommandLineProcessor cmdp(false,true);
+    cmdp.setOption("verbose","quiet",&verbose,"Print messages and results.");
+    cmdp.setOption("debug","nondebug",&debug,"Print debugging information from solver.");
+    cmdp.setOption("pseudo","block",&pseudo,"Use pseudo-block or block GMRES solver.");
+    cmdp.setOption("frequency",&frequency,"Solvers frequency for printing residuals (#iters).");
+    cmdp.setOption("tol",&tol,"Relative residual tolerance used by GMRES solver.");
+    cmdp.setOption("num-rhs",&numrhs,"Number of right-hand sides to be solved for.");
+    cmdp.setOption("block-size",&blocksize,"Block size used by GMRES.");
+    cmdp.setOption("max-iters",&maxiters,"Maximum number of iterations per linear system (-1 = adapted to problem/block size).");
+    cmdp.setOption("max-subspace",&maxsubspace,"Maximum number of blocks the solver can use for the subspace.");
+    cmdp.setOption("max-restarts",&maxrestarts,"Maximum number of restarts allowed for GMRES solver.");
+    cmdp.setOption("nx",&nx,"Number of discretization points in each direction of 3D Laplacian.");
+    cmdp.setOption("ortho",&ortho,"Orthogonalization being used by GMRES solver.");
+    if (cmdp.parse(argc,argv) != Teuchos::CommandLineProcessor::PARSE_SUCCESSFUL) {
+      return -1;
+    }
+    if (!verbose)
+      frequency = -1;  // reset frequency if test is not verbose
+
+    proc_verbose = ( verbose && (MyPID==0) ); // Only print on the zero processor
+
+    if (proc_verbose) {
+      std::cout << Belos::Belos_Version() << std::endl << std::endl;
+    }
+
+    //
+    // Set up the test problem.
+    //
+    // We use Trilinos' Galeri package to construct a test problem.
+    // Here, we use a discretization of the 2-D Laplacian operator.
+    // The global mesh size is nx * nx.
+    //
+
+    Teuchos::ParameterList GaleriList;
+    GaleriList.set ("n", nx * nx * nx);
+    GaleriList.set ("nx", nx);
+    GaleriList.set ("ny", nx);
+    GaleriList.set ("nz", nx);
+
+    auto Map = RCP{Galeri::Xpetra::CreateMap<ST,GO,tmap_t>("Cartesian3D", Comm, GaleriList)};
+    auto GaleriProblem = Galeri::Xpetra::BuildProblem<ST,LO,GO,tmap_t,tcrsmatrix_t,MV>("Laplace3D", Map, GaleriList);
+
+    // Create matrix from problem
+    auto A = GaleriProblem->BuildMatrix();
```

Comment:
I think that for matrix names variable naming can be uppercase because it is some mathematical naming
so this is good here I think

Reply:
Resolved